src/console.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * This file is part of Privacy Badger <https://www.eff.org/privacybadger>
+ * Copyright (C) 2015 Electronic Frontier Foundation
+ *
+ * Privacy Badger is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Privacy Badger is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Privacy Badger.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+function logMessage () {
+  var messages = [];
+  var _log = debounce(function () {
+    setTimeout(function () {
```

Comment:
My `setTimeout` suggestion was meant to help avoid blocking web request processing inside the `chrome.webRequest` listeners. Any "blocking" listeners are especially performance critical. Any slow code there holds up page loading for the user. From https://developer.chrome.com/extensions/webRequest:
>If the optional opt_extraInfoSpec array contains the string 'blocking' (only allowed for specific events), the callback function is handled synchronously. That means that the request is blocked until the callback function returns.